### PR TITLE
Added PubKeyHash type support

### PR DESCRIPTION
**Pull request by msinkec**

### CODE REVIEW

**Review by xhliu — COMMENTED** [review]
👍 looks good.

Let us add an e2e test here so we can be sure it works, w/o relying on another repo like boilerplate. Refer to alias.test.ts for examples.

**Comment by msinkec** [comment]
OK, I've added the test. I've named the test file domainSubtypes because PubKeyHash behaves more as a separate type as an alias and also the desc file has no info about it being an alias.

https://github.com/msinkec/scryptlib/commit/21b89a34e92843d094d262fb8a7d2f374369e5ee

File: test/fixture/desc/p2pkh_desc_w_pubkeyhash_type.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
```
Comment (xhliu) [inline test/fixture/desc/p2pkh_desc_w_pubkeyhash_type.json:1]: Let's not check in json file, unless absolutely necessary.

**Review by xhliu — APPROVED** [review]
:shipit: good to go.

File: test/fixture/desc/p2pkh_desc_w_pubkeyhash_type.json (hunk 1 above)
Comment (msinkec) [reply test/fixture/desc/p2pkh_desc_w_pubkeyhash_type.json:1]: It's needed in the test for PubKeyHash. Would it be better to compile the sCrypt source file itself in the test?

**Comment by zhfnjust** [comment]
another pr fix this https://github.com/sCrypt-Inc/scryptlib/pull/170

**Pull request closed without merging**